src/main/java/modbus/ModbusLink.java

```diff
@@ -374,17 +370,9 @@ private void restoreLastSession() {
 				if (transType != null && host != null && port != null && maxrbc != null && maxrrc != null
 						&& maxwrc != null && ddd != null && mwo != null && slaveId != null && interval != null
 						&& timeout != null && retries != null) {
-					String hostName = host + ":" + port;
-					IpConnection conn = null;
-					if (hostToConnection.containsKey(hostName)) {
-						conn = hostToConnection.get(hostName);
-					} else {
-						conn = new IpConnection(getLink(), child);
-						hostToConnection.put(hostName, conn);
-					}
-
-					SlaveNodeWithConnection sn = new SlaveNodeWithConnection(getLink(), conn, child);
-					sn.restoreLastSession();
+
+					ModbusConnection conn = new IpConnectionWithDevice(getLink(), child);

```

Comment:
The restore last session is restoring a "folder", but you create an IpConnectionWithDevice instead.